### PR TITLE
Raw order picking config

### PR DESCRIPTION
**Pull request by bbrito**
requires https://github.com/ipa320/cob_common/pull/207

Actual configuration of RAW with UR arm

### CODE REVIEW

**Comment by fmessmer** [comment]
This has a merge conflict...as we did some bigger restructuring in cob_robots...please merge with latest cob_robots/indigo_dev, adjust accordingly and test updated branch on robot..

**Comment by fmessmer** [comment]
Your merge didn't go well....there is file which are already merged into ipa320/indigo_dev and you commited a lot of temp files from merge conflicts, e.g. *.orig or *.BACKUP.*, *.BASE.*, *.LOCAL.*

**Comment by bbrito** [comment]
The masses of the UR10 arm should be reduced in order to get a stable simulation. The robot configuration  is now fine.

**Comment by fmessmer** [comment]
Has this been tested again with hardware?

**Comment by fmessmer** [comment]
> The masses of the UR10 arm should be reduced in order to get a stable simulation. The robot configuration is now fine.

Can you try [this](https://github.com/ipa320/cob_common/pull/207#issuecomment-279710744) tweaking the base mass/inertia so that we do not rely on modified `ur_description`?

**Comment by bbrito** [comment]
Done. I icreased the batery mass to 50 kg and it worked

File: cob_hardware_config/raw3-1/config/arm_controller.yaml [hunk 1]
```diff
@@ -32,4 +32,77 @@ joint_trajectory_controller:
...
+  required_drive_mode: 1
+arm_6_joint_position_controller:
+  type: position_controllers/JointPositionController
+  joint: arm_wrist_3_joint
+  required_drive_mode: 1
+
+
+## velocity controller
+joint_group_velocity_controller:
+  type: velocity_controllers/JointGroupVelocityController
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_controller.yaml]: note that currently [ur_description](https://github.com/ros-industrial/universal_robot/blob/indigo-devel/ur_description/urdf/ur.transmission.xacro) only supports `JointPositionInterface`, i.e. `position_controllers`

File: cob_hardware_config/raw3-1/config/arm_self_collision.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## The following parameter structure defines what pairs of links are to be considered during self-collision avoidance
+
+## Left: The link that is considered an 'obstacle' during self-collision avoidance
+## Right: The links of the component that are to be checked for self-collision with the 'obstacle'
+self_collision_map: {
+    ##head_3_link : [arm_left_3_link, arm_left_4_link, arm_left_5_link, arm_left_6_link, arm_left_7_link],
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_self_collision.yaml]: leave file empty...or set proper link_names...but those commented are not available on raw3-1 with ur....

File: cob_hardware_config/raw3-1/urdf/raw3-1.urdf.xacro [hunk 3]
```diff
@@ -42,174 +44,12 @@
     <origin xyz="${head_x} ${head_y} ${head_z}" rpy="${head_roll} ${head_pitch} ${head_yaw}" />
   </xacro:raw_head>
 
-<!--
   <xacro:ur10 name="arm" parent="base_link">
     <origin xyz="${arm_x} ${arm_y} ${arm_z}" rpy="${arm_roll} ${arm_pitch} ${arm_yaw}" />
   </xacro:ur10>
 
-  <xacro:raw_boxgripper name="gripper" parent="arm_ee_link">
+  <xacro:vacuum_gripper_definition name="gripper" parent="arm_ee_link" prefix="${arm_prefix}">
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/urdf/raw3-1.urdf.xacro]: macro name has changed in cob_common....`vacuum_gripper`....

Comment (fmessmer) [inline cob_hardware_config/raw3-1/urdf/raw3-1.urdf.xacro]: `prefix` parameter has been removed...

File: cob_hardware_config/raw3-1/urdf/raw3-1.urdf.xacro [hunk 4]
```diff
@@ -42,174 +44,12 @@
...
-    <visual>
-      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
-      <geometry>
-        <box size="0.10 0.03 0.03" />
-      </geometry>
-      <material name="IPA/Black" />
-    </visual>
-  </link>
-  <!-- END: Kinect mount -->
+  </xacro:vacuum_gripper_definition>
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/urdf/raw3-1.urdf.xacro]: macro name!

**Review by fmessmer — CHANGES_REQUESTED** [review]

File: cob_hardware_config/raw3-1/config/arm_controller.yaml (hunk 1 above)
Comment (bbrito) [reply cob_hardware_config/raw3-1/config/arm_controller.yaml]: Sorry I think i mix this with other branch. I'm going to test the new driver to use velocity control for this arm but I will submit this if this works in a future pull request.

File: cob_hardware_config/raw3-1/config/arm_cartesian_controller.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
+## Cartesian controller parameters
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_cartesian_controller.yaml]: so this file should be removed from this PR, then

File: cob_hardware_config/raw3-1/config/arm_controller.yaml [hunk 6]
```diff
@@ -32,4 +32,39 @@ joint_trajectory_controller:
   state_publish_rate:  25
   action_monitor_rate: 10
 
-## no further controllers available
+## position controller
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_controller.yaml]: these also won't work with the real hardware

File: cob_hardware_config/raw3-1/config/arm_self_collision.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+## The following parameter structure defines what pairs of links are to be considered during self-collision avoidance
+
+## Left: The link that is considered an 'obstacle' during self-collision avoidance
+## Right: The links of the component that are to be checked for self-collision with the 'obstacle'
+self_collision_map: {
+    ##head_3_link : [arm_shoulder_pan_joint, arm_shoulder_lift_joint, arm_elbow_joint, arm_wrist_1_joint, arm_wrist_2_joint, arm_wrist_3_joint],
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/arm_self_collision.yaml]: there is no `head_3_link` on raw3-1

File: cob_moveit_config/raw3-1/config/ompl_planning.yaml [hunk 8]
```diff
@@ -1,75 +1,68 @@
 planner_configs:
```
Comment (fmessmer) [inline cob_moveit_config/raw3-1/config/ompl_planning.yaml:1]: did you also update the SRDF?
you can use `rosrun cob_moveit_bringup update_collisions.sh raw3-1` to update the self-collision matrix
please also update the planning groups!

**Review by fmessmer — CHANGES_REQUESTED** [review]
And I see the following warnings:
 > [ WARN] [1487232228.326437512]: Parameter 'devicedriver' is missing. Using default Value: cob_ledboard

 > [ WARN] [1487232228.471580320]: Parameter 'num_leds' is missing. Using default Value: 58

File: cob_moveit_config/raw3-1/config/ompl_planning.yaml (hunk 8 above)
Comment (bbrito) [reply cob_moveit_config/raw3-1/config/ompl_planning.yaml:1]: Well this was updated and tested from the vision fair. So it should still be updated if this configuration was not lost with all these last changes in the repository.

**Comment by bbrito** [comment]
I have no idea how to fix these warning. Do you?

File: cob_hardware_config/raw3-1/config/light.yaml [hunk 9]
```diff
@@ -1,7 +1,9 @@
+devicedriver: ms-35
 devicestring: /dev/ttyUSB0
-baudrate: 230400
-invert_output: true
+baudrate: 38400
+num_leds: 58
+invert_output: false
```
Comment (fmessmer) [inline cob_hardware_config/raw3-1/config/light.yaml:5]: @ipa-bnm 
Could you please have a look at this yaml?
We used the layout from raw3-3...do you know what value to set for `devicedriver`, `baudrate` and `num_leds` for raw3-1?

**Review by fmessmer — APPROVED** [review]

**Comment by fmessmer** [comment]
@ipa-bfb 
I had to comment the `ur_modern_driver` part in https://github.com/ipa320/cob_robots/pull/601/commits/115c020ff201206dfd081e793505cea2c842e0a3 in order to pass the CI-Tests...
We will need to deal with this in a separate PR when you  did further tests...

**Comment by bbrito** [comment]
Sure ;)

**Pull request merged**